doc/prelimsLinux.rst

```diff
@@ -0,0 +1,222 @@
+
+.. code-block:: shell
+
+   cmake --version
+
+.. warning:: 
+
+   Depending on the OS being run, a repository update may be neccesary to install the newer version of CMake.
+
+b) Install Armadillo
+********************
+
+`Armadillo <http://arma.sourceforge.net/>`__ is a wrapper that presents a clean interface to several linear algebra libraries. To install Armadillo, it is best to follow the instructions outlined `here <http://arma.sourceforge.net/download.html>`__. First, install the reccomended packages based the OS being run. Then, in the command line, use wget to download the .tar archive and extract it (use the link above to find the most recent release):
+
+.. code-block:: shell
+   
+   wget http://sourceforge.net/projects/arma/files/armadillo-7.950.1.tar.xz
+   tar xzf armadillo-7.950.1.tar.xz
+
+The README.txt file found in the folder created by unpacking the Armadillo archive contains the remaining instructions. The most important of these is to change to the directory of the created folder and enter the following commands to configure armadillo:
+
+.. code-block:: shell
+
+   cmake .
+   make
+   make install   # sudo might be required
+
+To test that armadillo and its prerequisites have been installed correctly, run the included tester executable with the following commands:
+
+.. code-block:: shell
+
+   cd tests
+   make clean
+   make
+   ./main
+
+c) Install the HDF5 Library
+***************************
+
+The `HDF5 Library <https://support.hdfgroup.org/HDF5/>`__ (compiled with C++ support) is used for storing and managing raw experimental data. It is easiest to install and build the library with CMake, the steps for which can be found `here <https://support.hdfgroup.org/HDF5/release/cmakebuild518.html>`__. To download and uncompress the file, find the link to the most recent release and enter the following into the command line.
+
+.. code-block:: shell
+
+   wget https://support.hdfgroup.org/ftp/HDF5/current18/src/CMake-hdf5-1.8.19.tar.gz
+   tar xzf CMake-hdf5-1.8.19.tar.gz 
+
+Then change to the directory created by the extraction and execute the batch file named *build-unix.sh*.
+
+.. code-block:: shell
+
+   cd CMake-hdf5-1.8.19
+   ./build-unix.sh   # sudo might be required
+
+This will place the built binary in the bin folder and run through a series of tests for correct installation.
+
+d) Install and Compile mcopt Library
+************************************
+
+Finally, install the mcopt library itself; it can be found `here <https://github.com/jbradt/mcopt>`__. Clone the repository locally using the .git link found on GitHub.
+
+.. code-block:: shell
+
+   git clone https://github.com/jbradt/mcopt.git
+   cd mcopt
+
+The compilation and installation instructions can be found in the README.md file in this directory. The necessary commands are as follows.
+
+.. code-block:: shell
+
+   mkdir build && cd build
+   cmake -DCMAKE_BUILD_TYPE=Release ..
```

Comment:
You might also need `-DCMAKE_INSTALL_PREFIX` or `-DCMAKE_PREFIX_PATH` depending on where you want to install it (e.g. in your home directory on the HPCC). Maybe you could add a `.. note::` to tell people to look at the CMake documentation if they need to do fancy things like that.